File: Data-Structure/LinkedList/Linear Linkedlist/Singly LinkedList/Linkedlist.java
import java.util.Scanner;

// Node class
class Node{
	private int data;
	private Node next;

	public Node(int data) {
		this.data = data;
		this.next = null;
	}

	// Getters
	public int getData() { return data; }
	public Node getNext() { return next; }

	// Setters
	public void setNext(Node nextNode) { this.next = nextNode; }
}


class SimplyLinkedlist {
	// List variables
	private int LENGTH = 0;
	private Node head, temp1, temp2;

	// Get length of list method
	public int getLength() {
		return LENGTH;
	}

	// Insert Node
	public void insert (int index, int data) {
		if (LENGTH==0 && index==1) {  // If list is empty
			Node newNode = new Node(data);  // Create temp2
			head = newNode;
			LENGTH++;
			System.out.println(data + " endered at in begining.");
		} else if (index>0 && index<=LENGTH+1) {
			Node newNode = new Node(data);  // Create temp2
	
			if (index == 1) {   // If index is head
				newNode.setNext(head);
				head = newNode;
			} else {      // Other position in list
				temp1 = head;
				for (int i=1; i<index-1; i++)   // Traverse to index-1 node
					temp1 = temp1.getNext();
				// Making links
				newNode.setNext(temp1.getNext());
				temp1.setNext(newNode);
			}
	
			LENGTH++;
			System.out.println(data + " endered at " + index + ".");
		} else {
			System.out.println("Invalid Index");
		}
	}

	// Delete Node
	public void delete(int index) {
		if (LENGTH == 0) {
			System.out.println("Empty List.");
		} else {
			if (index>0 && index<=LENGTH) {

				if (index == 1) {   // If index is head
					temp2 = head;
					temp1 = temp2.getNext();
					head = temp1;
				} else {
					temp1 = head;
					for (int i=1; i<index-1; i++)   // Traverse to index-1 node
						temp1 = temp1.getNext();
					// Removing links
					temp2 = temp1.getNext();
					temp1.setNext(temp2.getNext());
				}

				System.out.println(temp2.getData() + " is deleted from list.");
				LENGTH--;
			} else {
				System.out.println("Invalid Index.");
			}
		}
	}

	// Display
	public void display() {
		if (LENGTH == 0) {
			System.out.println("Empty list.");
		} else {
			temp1 = head;
			while (temp1 != null) {	// Traverse till end
				System.out.print(temp1.getData() + "\t");
				temp1 = temp1.getNext();
			}
			System.out.println();
		}
	}

}

public class Linkedlist {
	public static void main(String[] args) {
	
		// Object Initialization
		Scanner sc = new Scanner(System.in);
		SimplyLinkedlist ll = new SimplyLinkedlist();	

		int switch_var;		// Switch Variable
		int exit = 0;		// Exit condition
		int t1, t2;			// Temporary variable

		while (exit == 0) {
			// Menu
			System.out.print("\nEnter \n\t1. Insert \n\t2. Delete \n\t3. DISPLAY \n\t4. EXIT \n-> ");
			switch_var = sc.nextInt();

			switch (switch_var) {
				// INSERT
				case 1:
					System.out.print("Enter a value to insert : ");
					t1 = sc.nextInt();
					System.out.print("Enter  index(1-"+(ll.getLength()+1)+") : ");
					t2 = sc.nextInt();
					ll.insert(t2, t1);
					break;
				// DELETE
				case 2:
					System.out.print("Enter  index(1-"+ll.getLength()+") : ");
					t1 = sc.nextInt();
					ll.delete(t1);
					break;
				// DISPLAY
				case 3:
					ll.display();
					break;
				// EXIT
				case 4:
					exit = 1;
					break;
				// INVALID
				default:
					System.out.println("Enter a valid option.");
					break;
			}
		}
		sc.close();
	}
}

File: Data-Structure/LinkedList/Linear Linkedlist/Singly LinkedList/Linkedlist.py
class Createnode:
    def __init__(self, data):
        self.data = data
        self.next = None


class Linkedlist:
    # Initializing head
    def __init__(self):
        self.head = None
        self.length = 0

    # Inserting new node
    def insert(self, index , data):
        # If list is empty
        if self.length==0 and index==1:
            self.head = Createnode(data)
            self.length += 1
            print(f"{data} is inserted at begining of list.")
        # if list is not empty
        else:
            # For valid index
            if index>0 and index<=self.length+1:

                newNode = Createnode(data)  # Create node
                if index == 1:      # For inserting at position 1
                    newNode.next = self.head
                    self.head = newNode
                else:               # For remaining position
                    temp = self.head
                    for i in range(1, index-1):    # Traversing to index-1 node
                        temp = temp.next
                    newNode.next = temp.next
                    temp.next = newNode
                self.length += 1
                print(f"{data} is inserted at index {index}.")
                
            else:
                print("Invalid Index.")

    # Delete node
    def delete(self, index):
        if self.length == 0:    # if list is empty
            print("Empty list.")
        else:
            if index>0 and index<=self.length:

                if index == 1:      # For deleting at position 1
                    temp = self.head
                    self.head = temp.next
                else:               # For remaining position
                    temp1 = self.head
                    for i in range(1, index-1):    # Traversing to index-1 node
                        temp1 = temp1.next
                    temp = temp1.next
                    temp1.next = temp.next
                self.length -= 1
                print(f"{temp.data} is deleted from list.")
                
            else:
                print("Invalid Index.")

    # Traverse through list
    def display(self):
        if self.length == 0:    # if list is empty
            print("Empty list.")
        else:
            temp = self.head
            while temp != None:
                print(temp.data, end="   ")
                temp = temp.next


menu = '''
Enter
    1. Insert
    2. Delete
    3. Display
    4. Exit
-> '''
if __name__ == "__main__":
    obj = Linkedlist()
    exit = False
    while not exit:
        switch_var = int(input(menu))
        # INSERT
        if switch_var == 1:
            temp = int(input("Enter the number : "))
            index = int(input(f"Enter the index(1-{obj.length+1}) : "))
            obj.insert(index, temp)
        # DELETE
        elif switch_var == 2:
            index = int(input(f"Enter the index(1-{obj.length}) : "))
            obj.delete(index)
        # DISPLAY
        elif switch_var == 3:
            obj.display()
        # EXIT
        elif switch_var == 4:
            exit = True
        # INVALID
        else:
            print("Please enter a valid statement.")